en/guide/define_xml_element.md

```diff
@@ -0,0 +1,424 @@
+# How to Define MAVLink Messages & Enums
+
+MAVLink enums, messages, commands, and other elements are [defined within XML files](../messages/README.md) and then converted to libraries for [supported programming languages](../README.md#supported_languages) using a *code generator*.
+
+This topic provides practical guidance for defining and extending MAVLink XML elements, including conventions and best-practice.
+
+> **Note** For detailed information about the file format see [MAVLink XML Schema](../guide/xml_schema.md#messages) (you can also inspect [common.xml](https://github.com/mavlink/mavlink/tree/master/message_definitions/v1.0/common.xml) and other dialect files). 
+
+## Should we Define Messages or Commands?
+
+There are two ways to send information between MAVLink systems (including commands, information and acknowledgments):
+* [Messages](#messages) are encoded using `mavlink` elements. The message structure/fields and handling are largely unconstrained (i.e. up to the creator).
+* [MAVLink Commands](#mavlink_commands) are defined as entries in the [MAV_CMD](../messages/common.md#MAV_CMD) enum, and encoded into real messages that are sent using the [Mission Protocol](../protocol/mission.md) or [Command Protocol](../protocol/command.md). 
+  Their structure is defined (they have 7 `float` parameters *or* 5 `float` and 2 `int32_t` parameters) and handling/responses depend on the protocol used to send them.
+
+The guidance below provides some suggestions on when one or the other might be more appropriate.
+
+Consider using a proper message if:
+* The required information does not fit into a command (i.e. it requires more than 7 messages or the information cannot be stored in the available `float`/`int32_t` fields).
+* The message is part another protocol.
+
+Consider using a command if:
+* The message should be executed as part of a mission
+* There is an existing mission command that you wish to use outside of missions. 
+  Depending on the autopilot you may be able to handle the message using the same code for both modes.
+* You're working with MAVLink 1 and there is no free id for the new message
+  (MAVLink 1 has a much larger free pool of ids for MAVLink commands than for message ids).
+* Your command/message requires an ACK/NACK; using the existing protocol acknowledgments may be faster/easier than defining another message for acknowledgments.
+
+Otherwise either method may freely be used.
```

Comment:
Leaving size/fitting constraints aside I always thought of it this way:
A **command** is used to for sporadic events where you want to trigger something. It is important that the the command is not missed, hence the ack.
A **message** is used to stream a continuous flow of data (might be high or low frequency, but it is not a one time event). Some data might get lost but the receiving side will deal with that.

Reply:
Thanks @thomasgubler 

Hmmm, the actual division of existing messages/commands is not that clean. Protocols often provide acks and use messages directly (all the command protocol does for you is formalise that).  I guess you must use messages and an ack-less protocol for streams. 
You do need to use messages for broadcast streams, because command protocol assumes a destination.

Upshot, I think it is reasonable that these be added as additional reasons to each list above, rather than replacing them.

Does that make sense?

Reply:
yes

Reply:
Cool. Update added.